src/test/kotlin/uk/gov/justice/digital/hmpps/managerecallsapi/integration/IntegrationTestBase.kt

```diff
@@ -21,6 +23,7 @@ import java.util.concurrent.ThreadLocalRandom
 @SpringBootTest(webEnvironment = RANDOM_PORT)
 @ActiveProfiles("test")
 @TestInstance(TestInstance.Lifecycle.PER_CLASS)
+@AutoConfigureEmbeddedDatabase(provider = ZONKY)
```

Comment:
We don't need an embedded database for every integration test (e.g. `RecallsIntegrationTest`) so it would be preferable if annotation was only added for the appropriate tests.

Reply:
Not sure I agree with integration tests that mock out the database.  But I can remove the annotation.  Maybe we need a different TestBase then for ones that use a DB or not

Reply:
`RecallsIntegrationTest` is specifically testing the REST interface. We have another integration test `RecallRepositoryIntegrationTest` which tests the persistence layer using an embedded Postgres database. The advantage is that each of these tests focuses solely on one integration aspect. Also, `RecallsIntegrationTest` runs faster if it doesn't have to spin up a DB and it means you can test out scenario with the mock like if an exception is thrown by the repository and you expect a 500 response with a particular message - that's more tricky to do with an embedded database.